### DISCUSSION TITLE
My commits are unverified despite having public key added to GitHub

### DISCUSSION THREAD

**Discussion by mrpmartin** [community/community#106560] [New to GitHub] [answered]
Hello - I'm a bit of a newbie. Thanks in advance :)

All my commits are showing as 'unverified' now, and it tells me to upload the public key, which I've done. I just re-added the public key again, and made sure my Git Config file is updated properly to sign my commits. I tested committing a small change (using VS Code) and I was asked to auth SSH (I use 1Password) so I know the signing is working at least somewhere. 

Also, I can see that commits were previously 'verified' so I know it was working at some point. I haven't changed anything, but it's possible I did something inadvertently.

I'm not sure where this problem might be coming from now. Any thoughts on where a good starting point might be?

**Comment by Ariffazeel99** [comment]
If your commits on GitHub are showing as "unverified" even after you've added your public key, here are simplified steps to help fix the issue:

1. **Verify Your Email:** Make sure the email you use in your Git commits is the same one linked to your GitHub account and is verified.

2. **Use GPG for Signing:** Ensure you're using a GPG key for signing commits. SSH keys are for repository access, not for commit verification.

3. **Add Your GPG Key to GitHub:**
   - Generate a GPG key if you haven't already.
   - Add the GPG key to your GitHub account under "SSH and GPG keys" in settings.

4. **Configure Git to Sign Commits:**
   - Set your GPG key in Git with `git config --global user.signingkey YOUR_GPG_KEY_ID`.
   - Make sure your Git `user.email` and `user.name` match your GitHub account details.

5. **Sign Your Commits:**
   - When committing changes, use `git commit -S -m "Your message"` to sign your commit. In VS Code, check if there's a setting to automatically sign commits.

6. **Check Your Work:** After committing, use `git log --show-signature -1` to see if your commit is signed.

7. **Troubleshoot:** If commits are still unverified:
   - Double-check the GPG key is correctly added to GitHub.
   - Ensure your Git email configuration matches the email associated with your GitHub account and GPG key.

Remember, commit signing verifies your identity, making your contributions more secure. If you encounter issues, reviewing GitHub's documentation on GPG key setup and commit signing can provide additional guidance.

**Reply by Ariffazeel99** [reply]
you can share your issue with screenshot if you are still facing the problem thanks

**Reply by mrpmartin** [reply]
Hey @Ariffazeel99 thanks for all this helpful info.

I've confirmed the email in commits is the same as GitHub, and the GPG signing is handled by 1Password. And I actually think there's a problem with that 1Password service. After running `git log --show-signature -1` I'm getting this error:

```
error: gpg.ssh.allowedSignersFile needs to be configured and exist for ssh signature verification
commit 26c2b9daae397a6c4bb428adf1b746abcdcfbfab (HEAD -> main, origin/main, origin/HEAD)
No signature
```
I'll have to look into why that file is erroring. With 1Password they provide this snippet to add to the ~/.gitconfig file:
```
[user]
  signingkey = {public key}
[gpg]
  format = ssh
[gpg "ssh"]
  program = "/Applications/1Password.app/Contents/MacOS/op-ssh-sign"
[commit]
  gpgsign = true
```
Adding that manually or using the 'apply to gitconfig automatically' setting from 1Password, I think that's where the problem exists for me. I appreciate the troubleshooting steps all the same! :)

**Reply by mrpmartin** [reply, marked as answer]
Update: 

Found the solution via 1Password developer docs: https://developer.1password.com/docs/ssh/git-commit-signing/#step-2-register-your-public-key

It seems things weren't properly configured on my end with 1Password. Issue is solved for me. Thanks again.

**Comment by queenofcorgis** [comment]
@mrpmartin way to go on self-solving 🎉 Thanks for coming back to share the solution and marking the answer ✅

**Comment by c0nscience** [comment]
Hmm I just ran into the same issue. I have added my ssh key to github already and configured everything the way 1password suggested. Still github shows my commits as unverified and suggest that I should add my ssh key. Any ideas?
